cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -186,34 +209,96 @@ func generateMountsModeRecover(recoverySystem string) error {
 	return fmt.Errorf("recover mode mount generation not implemented yet")
 }
 
+// helper to mount a partition from the same disk as another previously mounted
+// partition
+func mountPartitionLabelFromSameDiskAsMountPoint(diskmountpoint, label, target string) error {
+	// get the disk for the mountpoint
+	disk, err := partition.DiskFromMountPoint(diskmountpoint)
+	if err != nil {
+		return err
+	}
+
+	// find the partition on that disk with the label
+	partuuid, err := disk.FindMatchingPartitionUUID(label)
+	if err != nil {
+		return err
+	}
+
+	// mount that and return
+	fmt.Fprintf(stdout, "/dev/disk/by-partuuid/%s %s\n", partuuid, target)
+	return nil
+}
+
 func generateMountsModeRun() error {
-	// 1.1 always ensure basic partitions are mounted
-	for _, d := range []string{boot.InitramfsUbuntuSeedDir, boot.InitramfsUbuntuBootDir} {
-		isMounted, err := osutilIsMounted(d)
-		if err != nil {
-			return err
-		}
-		if !isMounted {
-			fmt.Fprintf(stdout, "/dev/disk/by-label/%s %s\n", filepath.Base(d), d)
-		}
+	// 1.1 mount ubuntu-boot first, using the LoaderDevicePartUUID to identify
+	//     the partuiid partition that should be mounted
+	//     in run mode, we should be getting our kernel from ubuntu-boot
+
+	isMounted, err := osutilIsMounted(boot.InitramfsUbuntuBootDir)
+	if err != nil {
+		return err
+	}
+	if !isMounted {
+		return selectPartitionMatchingKernelDisk(boot.InitramfsUbuntuBootDir, "ubuntu-boot")
```

Comment:
the tests seems not to get here

Reply:
hmm you're right I thought I had that tested but it seems I missed it

Reply:
this is fixed